common/internal_model/src/main/scala/uk/ac/wellcome/models/parse/PeriodParser.scala

```diff
@@ -166,6 +178,34 @@ object PeriodParser extends Parser[InstantRange] with DateParserUtils {
   def yearDivision[_: P]: P[FuzzyDateRange[MonthAndYear, MonthAndYear]] =
     seasonYear | lawTermYear
 
+  def calendarDate[_: P]: P[CalendarDate] =
+    numericDate | dayMonthYear | monthDayYear | yearMonthDay
+
+  def numericDate[_: P]: P[CalendarDate] =
+    (dayDigits ~ "/" ~ monthDigits ~ "/" ~ yearDigits)
+      .map { case (d, m, y) => CalendarDate(d, m, y) }
+
+  def dayMonthYear[_: P]: P[CalendarDate] =
+    (writtenDay ~ ws ~ (writtenMonth | monthDigits) ~ ",".? ~ ws ~ yearDigits)
+      .map { case (d, m, y) => CalendarDate(d, m, y) }
+
+  def monthDayYear[_: P]: P[CalendarDate] =
+    (writtenMonth ~ ws ~ writtenDay ~ ",".? ~ ws ~ yearDigits)
+      .map { case (m, d, y) => CalendarDate(d, m, y) }
+
+  def yearMonthDay[_: P]: P[CalendarDate] =
+    (yearDigits ~ ws ~ (writtenMonth | monthDigits) ~ ws ~ writtenDay).map {
+      case (y, m, d) => CalendarDate(d, m, y)
+    }
+
+  def monthAndYear[_: P]: P[MonthAndYear] =
+    (monthFollowedByYear | yearFollowedByMonth)
+      .map { case (m, y) => MonthAndYear(m, y) }
+
+  def month[_: P]: P[Month] = writtenMonth map Month
+
+  def day[_: P]: P[Day] = writtenDay map Day
```

Comment:
Note to self: these methods have all been moved unmodified from FreeformDateParser.